plugins/modules/domain.py

```diff
@@ -99,65 +106,60 @@ def run_module():
                     record = client.get(
                         '/domain/zone/%s/record/%s' % (domain, ind)
                     )
-                    # The record already exist
-                    if record['subDomain'] == name and record['target'] == value:
-                        module.exit_json(
-                            msg="{} is already registered on domain {}".format(name, domain),
-                            changed=False)
+                    # the record already exists : dont add it later
+                    if record['target'] in value:
+                        value.remove(record['target'])
+
+                    # the record must not exist and append is set to false : delete from zone
+                    if record['target'] not in value and not append:
```

Comment:
A bit worried to call "delete" endpoint in a "present" loop, but I don't see a better way :thinking: